boolean_algebra/and_gate.py

```diff
@@ -29,7 +29,7 @@ def and_gate(input_1: int, input_2: int) -> int:
     >>> and_gate(1, 1)
     1
     """
-    return int(input_1 and input_2)
+    return int((input_1, input_2).count(1) == 2)
```

Comment:
gate = int((input_1, input_2).count(1) == 2)
return gate

The motivation for the change was not obvious. As it also doesn't make sense to perform so many operations in return, it's better to separate them.